docs/DeveloperGuide.md

```diff
@@ -1,34 +1,369 @@
-# Developer Guide
+# Developer Guide for trakCAL
 
-## Design & implementation
+The purpose of this developer guide is as reference for future collaborators of **trakCAL**.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+By: CS2113-T09-4    Since: September 2020   Licence: MIT
 
+<br>
 
-## Product scope
-### Target user profile
+## Table of Contents
+
+<br>
+
+* Table of Contents
+{:toc}
+
+## 1.0 Setting up
+
+### 1.1 Prerequisites
+ * Java 11 (can be download from [here](https://docs.aws.amazon.com/corretto/latest/corretto-11-ug/downloads-list.html))
+ * Intellij IDE
+ 
+<br>
+    
+### 1.2 Setting up:    
+* Download the latest **traKCAL** jar file ([here](https://github.com/AY2021S1-CS2113T-T09-4/tp/releases))
+* Copy jar file into an empty folder in a convenient location (eg. Desktop)
+* Open terminal and navigate directory to the folder containing jar file 
+* Enter the following command line to run program: ```java -jar trakCAL.jar```   
+* Look through the user guide for a full detailed explanation on the functionality of **trakCAL**
+    
+<br>
+<br>    
+    
+## 2.0 Design 
+
+### 2.1 Architecture
+
+![Architecture](diagrams/Architecture.png)
+
+*Figure 1. Diagram for Model component*
+
+The Architecture Diagram shown above explains the high-level design of **trakCAL**.
+
+**trakCAL** is made up of mainly 6 components.
+
+`trakCAL`: tracKCAL starts here
+
+`Ui`: Displays any message that the user can see and interact with
+
+`Logic`: Interprets what the user input
+
+`Command`: Executes specific command according to interpretation by `Logic`
+
+`Storage`: Saves required data into the hard disk or retrieves data
+
+`Model`: Visualize data into graph
+
+<br>
+
+### 2.2 Logic component
+
+![Logic Component](diagrams/LogicComponent.png)
+
+*Figure 2. Diagram for logic component*
```

Comment:
![image](https://user-images.githubusercontent.com/46754292/97525243-563b6080-19e1-11eb-8a25-94e0c1ecd349.png)
Maybe you can change the arrowhead to another type to make it more clear that is an association.